### ISSUE TITLE
[Accessibility] Callout is not readable by windows narrator when callout is not having any focusable element

### ISSUE THREAD

**Issue by nikhilexe** [OfficeDev/office-ui-fabric-react#1655]
All callouts mentioned here https://dev.office.com/fabric#/components/callout are not readable by windows narrator. For example callout mentioned in the "Callout directional example" however "Nested callout..." content is readable by windows narrator.

Repro steps:

1) Launch edge and go to https://dev.office.com/fabric#/components/callout
2)  Press windows+Enter key to start narrator
3) Now make tab and reach to "Nested callout example..." and press enter
4) Narrator can read content by pressing CAPS + right/left arrow
5) Now move to "Callout directional example" and do the same...

Here narrator are not able to read the content and we can't move inside with CAPS + left/right arrow. The reason is that it doesn't have any focusable element. But ideally we should give way to read the content by Narrator, currently there is no way to do the same.

**Comment by betrue-final-final** [comment]
This should be investigated

**Comment by JasonGore** [comment]
I don't think this is an issue with Coachmark itself as much as these particular examples. As Coachmark is currently designed, accessibility would at least in part be the responsibility of the parent component. Even if we fix these examples, there is still the potential that any developer making use of Coachmark would make similar mistakes.

I think we need to explore design changes that would ensure consumers of Callout use Callout in a way that makes them accessible. ARIA itself does not seem to have a clear role that correlates with Callout's functionality. The closest role seems to be `tooltip`, but it is a role for a popup that is not interactive.

Coachmark has recently made changes in #5155 in an attempt to make it more accessible. Coachmark's functionality seems very similar to Callout in that it shows an popup to the user that may contain interactive content. Coachmark also makes use of TeachingBubble, which also has similar functionality.

I think this is a bigger task than just a bug fix with the following questions that should be answered:
- Should Coachmark, Callout and TeachingBubble all continue to be separate components?
    - Is Coachmark just a stylized version of Callout? 
    - Why are Callout and TeachingBubble separate components?
- Is there really no applicable ARIA role?
    - Coachmark is using a mix of `alert` and `dialog` roles while TeachingBubble also uses `option` roles. 
- How can we design Callout and similar components to make sure their parents make them accessible? (Via `aria-haspopup` attributes or similar.)
    - Coachmark in #5155 takes the approach of adding proprietary navigation behavior (via `ALT+C`) that allows users to focus the Coachmark content.

**Comment by dliu120** [comment]
Suggestion is for the callout to be focusable if it finds an element with tab-index=-1, so screen readers can read the content inside.

**Comment by philipzloh** [comment]
Jason hit the nail on the head. The issue here is not about focus. The issue here is that the callout needs a proper role in order for Narrator to read correctly. It is insufficient to just make items focusable with tabindex=-1.

The `tooltip` role does not imply non-interaction, in my interpretation.

**Comment by dzearing** [comment]
Do we have concrete scenarios to consider here where Callout would be used without any focusable content? And, in those scenarios, what SHOULD the screenreader read?

This sounds to me like a TeachingBubble scenario perhaps, but even then, I think TeachingBubbles have focus targets. The only scenario without a focus target I can come up with is Tooltip, but we surely would not want the content read in that case.

I think this really needs to be controlled by the app developer and not the utility for rendering callouts, and would rather close this than to add something scary like aria-live attributes which end up reading every tooltip you casually expose by moving the mouse.

**Comment by dzearing** [comment]
@cschlechty Just fyi here. Callout without focused content? Not sure the Callout itself is responsible for this scenario, but our options are: Make the container focusable? (Bad) Steal focus? (Bad) announce something? (please no)

**Comment by micahgodbolt** [comment]
@dzearing  what's wrong with the announced approach, but putting that on the user to supply the announcement when proving non actionable callouts? If a callout is nothing but visual information, the accessible equivalent is the announced control.

**Comment by JasonGore** [comment]
I think we're coming a bit full circle here, but unless there's a reason not to use the [ARIA `tooltip` role](https://www.w3.org/TR/wai-aria-1.1/#tooltip) we shouldn't try to create our own custom ARIA role using announced and live regions.

**Comment by micahgodbolt** [comment]
Okay, not sure about tooltip, but looking at the way Announced worked, we could treat the callout like announced does using role="status" and aria-live. It works well, can be done with existing control, and we could consider making this a prop to enable.

https://codepen.io/micahgodbolt/pen/abbxvPN?editors=0010

**Comment by dzearing** [comment]
A prop could work, but question; should these role/aria-live values be the default if not otherwise specified?

Users just need to know what to do. If they don't need to think about it, that's cool.

**Comment by micahgodbolt** [comment]
@dzearing we have two options to resolve this.

Keep the `AnnouncedCallout` component I created, or remove that extra export and simply show a code example of how you should use Callout with non focusable content.

Preference? Other devs are leaning towards not adding another exported control.

**Comment by micahgodbolt** [comment]
i pushed the code example we could give users to solve non-focusable content. If this is easy enough to use, we don't need to export a dedicated control for this.

https://github.com/OfficeDev/office-ui-fabric-react/pull/11357/files#diff-65937795be0311f788c0d880bfaab813R68-R84
